src/suggestion.rs

```diff
@@ -315,6 +315,26 @@ pub struct Suggestion<'s> {
     pub description: Option<String>,
 }
 
+impl<'s> Suggestion<'s> {
+    /// Determine if there is overlap.
+    pub fn is_overlapped(&self, other: &Self) -> bool {
+        if self.origin != other.origin {
+            return false;
+        }
+
+        if self.span.end.line < other.span.start.line || other.span.end.line < self.span.start.line
+        {
+            return false;
+        }
+
+        if self < other {
```

Comment:
```suggestion
        if self < other {
```

I don't quite understand this one? The implementation of `Ord` is rather complicated, so I'd prefer to be explicit here.

Reply:
@drahnr 
I wanted to inform you that I have made a change to explicitly compare the spans, rather than relying on the Ord implementation. I would greatly appreciate it if you could take a moment to review these changes. If everything looks good, I will proceed to update the test cases accordingly.